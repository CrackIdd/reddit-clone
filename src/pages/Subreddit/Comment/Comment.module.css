.comment {
    color: white;
    display: flex;
    column-gap: 8px;
}

.comment__content {
    display: flex;
    align-items: center;
}

.comment__author {
    margin-bottom: 6px;
    font-size: 12px;
    font-weight: 500;
    line-height: 16px;
}

.comment__message {

    font-size: 14px;
    margin-bottom: 6px;
}

.comment__treadline {
    height: 50px;
    border-right: 2px solid #343536;
    display: block;
    width: 50%;
}

.comment__icon {
    width: 36px;
    height: 36px;
    margin: 0 auto;
}

.comment__profile{
    display: flex;
    align-items: center;
    justify-content: center;
    width: 28px;
    height: 28px;
    border-radius: 50%;
    transform-origin: bottom center;
    background-color: #343536;
    margin-bottom: 12px;

}

.comment__author {

}