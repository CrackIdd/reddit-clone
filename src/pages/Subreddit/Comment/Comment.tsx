import { DocumentData } from "firebase/firestore";
import React from "react";
import CSSModules from "react-css-modules";
import styles from "./Comment.module.css";

const Comment: React.FC<DocumentData> = ({ comment }) => {
  return (
    <div styleName="comment">
      <div styleName="comment__author">{comment.content}</div>
    </div>
  );
};

export default CSSModules(Comment, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "log",
});
