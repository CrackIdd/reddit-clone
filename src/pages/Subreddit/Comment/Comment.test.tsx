import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import each from "jest-each";
import Comment from "./Comment";

describe("Comment", () => {
  const mockProps = {
    comment: {
      content: "this is a mock comment",
    },
  };
  test("snapshot", () => {
    const { asFragment } = render(<Comment {...mockProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
