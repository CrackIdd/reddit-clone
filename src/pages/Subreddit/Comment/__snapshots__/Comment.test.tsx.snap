// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment snapshot 1`] = `
<DocumentFragment>
  <div
    class="comment"
  >
    <div
      class="comment__author"
    >
      this is a mock comment
    </div>
  </div>
</DocumentFragment>
`;
